modules/apps/27-interchain-accounts/controller/keeper/msg_server.go

```diff
@@ -24,6 +26,10 @@ func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
 
 // RegisterInterchainAccount defines a rpc handler for MsgRegisterInterchainAccount
 func (s msgServer) RegisterInterchainAccount(goCtx context.Context, msg *types.MsgRegisterInterchainAccount) (*types.MsgRegisterInterchainAccountResponse, error) {
+	if msg == nil {
```

Comment:
wouldn't most of these be better placed in the `ValidateBasic` for each message?

Reply:
Good point, @DimitrisJim! Yes, you're right; the check is unnecessary. I think I saw it in one of the message handler and then I got carried away. :) I will revert now.